auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailActivity.java

```diff
@@ -111,28 +144,141 @@ private void setUpTermsOfService() {
+            } catch (IntentSender.SendIntentException e) {
+                Log.e(TAG, "Unable to start hint intent", e);
+            }
+        }
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        switch (requestCode) {
+            case RC_HINT:
+                if (data != null) {
+                    Credential credential = data.getParcelableExtra(Credential.EXTRA_KEY);
+                    if (credential != null) {
+                        mEmailEditText.setText(credential.getId());
+                        String name = credential.getName();
+                        mNameEditText.setText(name);
+                        if (TextUtils.isEmpty(name)) {
+                            mNameEditText.requestFocus();
+                        } else {
+                            mPasswordEditText.requestFocus();
+                        }
+                    }
+                }
+                break;
+            case RC_SIGN_IN:
+            case RC_WELCOME_BACK_IDP:
+                finish(resultCode, data);
+                break;
+        }
+    }
+
+    @Override
+    public void onFocusChange(View view, boolean hasFocus) {
+        if (hasFocus) return;
+
+        int id = view.getId();
+        if (id == R.id.email) {
+            String email = mEmailEditText.getText().toString();
+            if (mEmailFieldValidator.validate(mEmailEditText.getText())) {
+                if (!email.equals(mPrevEmail)) {
+                    mActivityHelper.showLoadingDialog(R.string.progress_dialog_checking_accounts);
+                    checkAccountExists(email);
+                    mPrevEmail = email;
+                }
+            }
+        } else if (id == R.id.name) {
+            mNameValidator.validate(mNameEditText.getText());
+        } else if (id == R.id.password) {
+            mPasswordFieldValidator.validate(mPasswordEditText.getText());
+        }
+    }
+
+    public void checkAccountExists(final String email) {
+        if (!TextUtils.isEmpty(email)) {
+            mActivityHelper.getFirebaseAuth()
+                    .fetchProvidersForEmail(email)
+                    .addOnFailureListener(
+                            new TaskFailureLogger(TAG, "Error fetching providers for email"))
+                    .addOnCompleteListener(
+                            new OnCompleteListener<ProviderQueryResult>() {
+                                @Override
+                                public void onComplete(@NonNull Task<ProviderQueryResult> task) {
+                                    if (task.isSuccessful()) {
+                                        List<String> providers = task.getResult().getProviders();
+                                        if (providers != null && !providers.isEmpty()) {
+                                            // Account does exist
```

Comment:
Recommended to expand comments to something like this:
```java
// There is an account tied to this email.  If only the email provider is associated with the account, direct the user
// to the email sign in flow.  Otherwise, direct the user to sign in with the IDP they previously selected.
```
```